File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();

        for(int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }

        List<Integer>[] buckets = new List[nums.length + 1];
        for(int key : map.keySet()) {
            int frequency = map.get(key);
            if(buckets[frequency] == null) {
                buckets[frequency] = new ArrayList<>();
            }
            buckets[frequency].add(key);
        }

        int[] res = new int[k];
        int counter = 0;
        for(int i = buckets.length - 1; i >=0 && counter < k; i--) {
            if(buckets[i] != null) {
                for(int num : buckets[i]) {
                    res[counter++] = num;
                    if(counter == k) return res;
                }
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-5.cpp
#include <unordered_map>
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> map;
        bool flag = false;
        for(int num : nums) {
            if(map.find(num) == map.end()) {
                map[num] = 1;
            }
            else{
                flag = true;
                break;
            }
        }
        return flag;
    }
};